include/nccl_ofi_cuda.h

```diff
@@ -10,9 +10,21 @@
 extern "C" {
 #endif
 
+#if HAVE_CUDA
+
+#pragma weak cuDriverGetVersion
+#pragma weak cuPointerGetAttributes
+#pragma weak cuCtxGetDevice
+#pragma weak cuDeviceGetCount
+#pragma weak cuFlushGPUDirectRDMAWrites
```

Comment:
why?  the current way matches what NCCL does, which seems like a much better path forward.